android/Staccato_AN/app/src/main/java/com/on/staccato/presentation/main/MainActivity.kt

```diff
@@ -410,6 +418,29 @@ class MainActivity :
         return timeDuration / MILLISECONDS_TO_SECONDS
     }
 
+    private fun registerAndLaunchNotificationPermission() {
+        notificationPermissionManager.register(
+            caller = this,
+            activity = this,
+            showRationale = ::showNotificationPermissionRationale,
+        )
+        notificationPermissionManager.launch()
+    }
+
+    private fun showNotificationPermissionRationale() {
+        binding.cvNotificationRationaleDialog.setContent {
+            NotificationPermissionRationale(::moveToNotificationSetting)
+        }
+    }
+
+    private fun moveToNotificationSetting() {
+        val intent =
+            Intent(Settings.ACTION_APP_NOTIFICATION_SETTINGS).apply {
+                putExtra(Settings.EXTRA_APP_PACKAGE, packageName)
+            }
```

Comment:
Intent의 설정 함수들은 자기 자신을 반환하므로 체이닝이 가능합니다! 설정값이 단순하므로 apply는 제거해도 좋을 것 같습니다~

```suggestion
        val intent =
            Intent(Settings.ACTION_APP_NOTIFICATION_SETTINGS)
                .putExtra(Settings.EXTRA_APP_PACKAGE, packageName)
```

Reply:
아 이거 맨날 까먹네여 ㅋㅋㅋ 반영 완